main.tf

```diff
@@ -0,0 +1,450 @@
+locals {
+  common_tags = {
+    itse_app_env      = var.itse_app_env
+    workspace         = var.environment
+    itse_app_customer = var.customer
+    managed_by        = var.managed_by
+    itse_app_name     = var.app_name
+    environment       = var.environment
+  }
+
+  # Use existing bucket name
+  backup_bucket_name = var.backup_bucket_name
+
+  # Use table names directly from variable
+  table_names = var.dynamodb_tables
+
+  # B2 environment variables (only set if B2 backup is enabled and credentials are provided)
+  b2_env_vars = var.b2_backup_enabled && var.b2_application_key_id != "" && var.b2_application_key != "" && var.b2_bucket != "" && var.b2_endpoint != "" ? {
+    B2_APPLICATION_KEY_ID = var.b2_application_key_id
+    B2_APPLICATION_KEY    = var.b2_application_key
+    B2_BUCKET             = var.b2_bucket
+    B2_ENDPOINT           = var.b2_endpoint
+  } : {}
+}
+
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+
+# Reference existing S3 bucket (created manually)
+data "aws_s3_bucket" "mfa_backups" {
```

Comment:
Why is this a `data` block and not a full `resource`? It feels awkward to combine this `data` with actual resources that depend on it (e.g. `aws_s3_bucket_versioning`).

Reply:
I haven't read the PR yet (my apologies), but I think it comes back to the question of whether backups should be in an S3 bucket created by Terraform (and thus, that could be destroyed by Terraform, such as in a DR exercise) or created manually and simply configured by Terraform. I lean towards creating the S3 bucket (for backups) manually outside of Terraform, but I don't remember if we have established some team-convention about this. Thoughts?